src/rules/order.js

```diff
@@ -526,8 +526,13 @@ function computeRank(context, ranks, importEntry, excludedImportTypes) {
   if (!excludedImportTypes.has(impType)) {
     rank = computePathRank(ranks.groups, ranks.pathGroups, importEntry.value, ranks.maxPosition);
   }
-  if (typeof rank === 'undefined') {
+
+  if (rank === undefined) {
     rank = ranks.groups[impType];
+
+    if(rank === undefined) {
```

Comment:
```suggestion
  if (typeof rank === 'undefined') {
    rank = ranks.groups[impType];

    if (typeof rank === 'undefined') {
```

Reply:
I hoped I could factor this change out from contrib-sort-types-among-themselves, but I've noticed the test needs `sortTypesGroup` to trigger the bad behavior :)